source/routes/guesses.js

```diff
@@ -0,0 +1,117 @@
+import { Router } from 'express';
+import { Guess } from '../models/guess.js';
+import { Game } from './../models/game.js'; 
+
+const guessRouter = Router();
+
+guessRouter.get('/:guessID', async (req, res) => {
+    const guessID = parseInt(req.params.guessID);
+ 
+    if (!guessID) {
+      return res.status(400).json({ status: 'fail', message: 'Guess Id is required!'});
+    }
+
+    const guess = await getGuessesById(guessID);
+  
+    if (guess) {
+      res.status(200).json({ status: 'success', guess});
+    } else {
+      res.status(404).json({ status: 'fail', message: 'No guess found!' });
+    }
+});
+
+
+guessRouter.get('/by-game/:gameID', async (req, res) => {
+    const gameID = parseInt(req.params.gameID);
+ 
+    if (!gameID || !await Game.query().findById(gameID)) {
+      return res.status(400).json({ status: 'fail', message: 'Invalid gameId chek if game with that id exists!'});
+    }
+
+    const guesses = await getAllGuessesByGameId(gameID);
+  
+    if (guesses) {
+      res.status(200).json({ status: 'success', guesses });
+    } else {
+      res.status(404).json({ status: 'fail', message: 'Error while getting guesses!' });
+    }
+});
+
+
+guessRouter.post("/create/:gameID", async(req,res) => 
+{
+  const guess = String(req.body["guess"]);
+  const gameID = parseInt(req.params.gameID);
+  
+  if (!/^(?!.*(.).*\1)\d{4}$/.test(guess)) 
+  {
+    return res.status(400).json({ status: 'fail', message: 'Error no guess or invalid guess provided!'});
+  }
+  
+  if (!gameID) {
+    return res.status(400).json({ status: 'fail', message: 'Invalid gameId!'});
+  }
+
+  const game = await Game.query().findById(gameID);
+
+  if(!game)
+  {
+    return res.status(400).json({ status: 'fail', message: 'No game with that Id exists!'});
+  } 
+
+  let animals = checkBullsAndCows(game,guess);
+  let hasGuessed = animals.bullsCount === 4;
```

Comment:
We can move the number in a constant. It is a bad practice to have so called magical numbers thrown around the code.